ims/einstein/operations.py

```diff
@@ -211,119 +207,108 @@ def shutdown(self):
+            # the copy in ceph of img_name
             ceph_img_name = self.__get_ceph_image_name(img_name)
+
+            # Prepare the ceph image name to be used.
+            clone_ceph_name = self.__get_ceph_image_name(disk_name)
+        except DBException as e:
+            logger.exception('')
+            return self.__return_error(e)
+
+        # Storage Operations
+        try:
             self.fs.clone(ceph_img_name, constants.DEFAULT_SNAPSHOT_NAME,
                           clone_ceph_name)
-            ceph_config = self.cfg.fs
-            logger.debug("Contents of ceph_config = %s", str(ceph_config))
-            self.iscsi.add_target(clone_ceph_name)
-            logger.info("The create command was executed successfully")
-            self.__register(node_name, img_name, clone_ceph_name, mac_addr)
-            return self.__return_success(True)
-
-        except RegistrationFailedException as e:
-            # Message is being handled by custom formatter
-            # TODO: add a deployment and a unit test for this case.
+        except FileSystemException as e:
             logger.exception('')
-            clone_ceph_name = self.__get_ceph_image_name(node_name)
-            self.iscsi.remove_target(clone_ceph_name)
-            self.fs.remove(clone_ceph_name)
             self.db.image.delete_with_name_from_project(node_name, self.proj)
-            time.sleep(constants.HIL_CALL_TIMEOUT)
-            self.hil.detach_node_from_project_network(node_name, network,
-                                                      nic)
-            return self.__return_error(e)
 
+        # iSCSI Operations
+        try:
+            self.iscsi.add_target(clone_ceph_name)
         except ISCSIException as e:
-            # Message is being handled by custom formatter
             logger.exception('')
-            clone_ceph_name = self.__get_ceph_image_name(node_name)
             self.fs.remove(clone_ceph_name)
             self.db.image.delete_with_name_from_project(node_name, self.proj)
-            time.sleep(constants.HIL_CALL_TIMEOUT)
-            self.hil.detach_node_from_project_network(node_name, network,
-                                                      nic)
-            return self.__return_error(e)
 
-        except FileSystemException as e:
-            # Message is being handled by custom formatter
-            logger.exception('')
-            self.db.image.delete_with_name_from_project(node_name, self.proj)
-            time.sleep(constants.HIL_CALL_TIMEOUT)
-            self.hil.detach_node_from_project_network(node_name, network,
-                                                      nic)
-            return self.__return_error(e)
+        logger.info("The create_disk command was executed successfully")
+
+        # This will be changed to return a list of all targets, with each
+        # target including the endpoint, port and target name.
+        return self.__return_success(clone_ceph_name)
```

Comment:
@apoorvemohan @Izhmash I am thinking of returning something like this
```python
>>> returnee = {"iscsi-server": "127.0.0.1","port": "3260", "target": "mytarget"}, {"iscsi-server": "192.168.3.1", "port": "3261", "target": "mytarget"}
>>> print returnee
({'target': 'mytarget', 'iscsi-server': '127.0.0.1', 'port': '3260'}, {'target': 'mytarget', 'iscsi-server': '192.168.3.1', 'port': '3261'})
>>> json.dumps(returnee)
'[{"target": "mytarget", "iscsi-server": "127.0.0.1", "port": "3260"}, {"target": "mytarget", "iscsi-server": "192.168.3.1", "port": "3261"}]'
```

Reply:
Unless there would be a case where the target dictionaries need to be named, I think this works well.